### PR TITLE
Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by IEvangelist**
Enable CodeQL on dotnet/docs-actions

### CODE REVIEW

**Comment by Youssef1313** [comment]
I'll take care of the failure here.

**Review by gewarren — APPROVED** [review]

**Comment by Youssef1313** [comment]
Fixed the dogfood check failure in https://github.com/dotnet/docs-actions/pull/9.

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ main, live ]
```
Comment (Youssef1313) [inline .github/workflows/codeql-analysis.yml:16]: Do we need a `live` branch in the repository at all (not talking about the yml file only, but the repo itself)?

Comment (IEvangelist) [reply .github/workflows/codeql-analysis.yml:16]: I defer to @gewarren -- but I doubt that we do.

Comment (gewarren) [reply .github/workflows/codeql-analysis.yml:16]: I don't see why we'd need a live branch.

**Pull request merged**